src/libraries/drand/_index.md

```diff
@@ -0,0 +1,118 @@
+
+On initialization, the Filecoin node can call the `Group` endpoint in order to obtain the beacon's [group file](https://github.com/drand/drand/blob/57a6056a24d4ebaa27a44852636807364624b9fc/key/group.go). The client (node) should then have the hash of the group file (or the full file itself if it already has it) cached to verify the group it receives corresponds to the expected one. It should also cache:
+
+- cache the beacon's `Period`                           -- the period of time between each drand randomness generation
+- cache the beacon's `GenesisTime`                      -- at which the first round in the drand randomness chain is created
+- verify that the beacon's `PublicKey` is appropriate   -- ie that the filecoin node connected to the right drand beacon
+
+Thereafter, the Filecoin client can call drand's endpoints:
+
+- `LastPublic` to get the latest randomness value produced by the beacon
+- `Public` to get the randoomness value produced by the beacon at a given index
+
+{{<label drand>}}
+### Using drand in Filecoin
+
+Drand is used as a randomness beacon for leader election in Filecoin. You can read more about that in {{<sref leader_election>}}. See drand used in the Filecoin lotus implementation [here](https://github.com/filecoin-project/lotus/blob/master/chain/beacon/drand/drand.go).
+
+While drand returns multiple values with every call to the beacon (see above), Filecoin blocks need only store a subset of these in order to track a full drand chain. This information can then be mixed with on-chain data for use in Filecoin. See {{<sref randomness>}} for more.
+
+#### Verifying an incoming drand value
+
+Upon receiving a new drand randomness value from a beacon, a Filecoin node should immediately verify its validity. That is, it should verify:
+
+- that the `Signature` field is verified by the beacon's `PublicKey` as the beacon's signature of `SHA256(PreviousSignature || Round)`.
+- that the `Randomness` field is `SHA256(Signature)`.
+
+See [drand](https://github.com/drand/drand/blob/master/beacon/beacon.go#L63) for an example.
+
+#### Fetching the appropriate drand value while mining
+
+There is a deterministic mapping between a needed drand round number and a Filecoin epoch number.
+
+After initializing access to a drand beacon, a Filecoin node should have access to the following values:
+
+- `filEpochDuration`    -- the Filecoin network's epoch duration (between any two leader elections)
+- `filGenesisTime`      -- the Filecoin genesis timestamp
+- `filEpoch`            -- the current Filecoin epoch
+- `drandGenesisTime`    -- drand's genesis timestamp
+- `drandPeriod`         -- drand's epoch duration (between any two randomness creations)
+
+Using the above, a Filecoin node can determine the appropriate drand round value to be used for use in {{<sref leader_election>}} in an epoch using both networks' reliance on real time as follows:
+
+```go
+MaxBeaconRoundForEpoch(filEpoch) {
```

Comment:
Why "Max"? this should just be BeaconRoundForEpoch, no?

Reply:
I ripped this from lotus mostly, but the rationale is this will return the highest such epoch number. In the case the beacon is faster than Filecoin block production there might be multiple rounds for an epoch. 

You can list these out by looking at the gap in function output between two rounds (same as in your doc fwiw).